File: Cap14VB/SearchRequestExtender.vb
Imports Cap14VB.Live.Search
Imports System.Reflection
Imports System.Text
Imports System.Globalization

Public Module SearchRequestExtender
  
  <System.Runtime.CompilerServices.Extension()> _
  Public Function TranslateToText(ByVal request As SearchRequest) As String
    If request Is Nothing Then
      Return ""
    End If

    Dim builder = New StringBuilder()
    Dim flags = BindingFlags.Instance Or BindingFlags.[Public]
    For Each prop In GetType(SearchRequest).GetProperties(flags)
      builder.AppendFormat("{0}: {1}" & vbLf, prop.Name, prop.GetValue(request, flags, Nothing, Nothing, CultureInfo.InvariantCulture))
    Next
    Return builder.ToString()
  End Function
End Module

File: Cap14CSharp/SearchRequestExtender.cs
using System;
using System.Globalization;
using System.Reflection;
using System.Text;
using Cap14CSharp.Live.Search;

namespace Cap14CSharp
{
    public static class SearchRequestExtender {
        public static String TranslateToText(this SearchRequest request) {
            if (request == null) return "";

            var builder = new StringBuilder();
            var flags = BindingFlags.Instance | BindingFlags.Public;
            foreach (var prop in typeof(SearchRequest).GetProperties(flags)) {
                builder.AppendFormat("{0}: {1}\n", prop.Name, 
                                     prop.GetValue(request,flags,null, null, CultureInfo.InvariantCulture));
            }
            return builder.ToString();
        }
    }
}